### PR TITLE
Maintenance: Remove sourcemaps generation

### PR DESCRIPTION
**Pull request by ndelangen**
## What I did

@yannbf & me noticed that around 6-12mb of static-storybook's size is sourcemaps.

When bindling our code (manager and prebuild) we should not generate sourcemaps, because of the size. We have `minifyIdentifiers` turned off:
https://github.com/storybookjs/storybook/blob/02e058a0d053b83055dc346f0c925231bb591155/scripts/prepare/bundle.ts#L190

... therefore sourcemaps do not add a lot to the ability to understand and debug the code.

Thus removing it is the best compromise.

Here is an example of before/after removing sourcemaps in the storybook test-runner example storybook statics:

Before:
```
472K    storybook-static/sb-preview
 16K    storybook-static/sb-addons/essentials-measure-6
 64K    storybook-static/sb-addons/ui-storybook-11
 56K    storybook-static/sb-addons/essentials-viewport-4
8.0K    storybook-static/sb-addons/links-0
1.1M    storybook-static/sb-addons/interactions-8
496K    storybook-static/sb-addons/designs-10
 16K    storybook-static/sb-addons/essentials-outline-7
 32K    storybook-static/sb-addons/essentials-toolbars-5
 52K    storybook-static/sb-addons/essentials-backgrounds-3
292K    storybook-static/sb-addons/essentials-actions-2
5.1M    storybook-static/sb-addons/storysource-9
2.0M    storybook-static/sb-addons/essentials-controls-1
9.3M    storybook-static/sb-addons
204K    storybook-static/sb-common-assets
3.4M    storybook-static/sb-manager
 19M    storybook-static/assets
 33M    storybook-static
```

After:
```
472K    ./storybook-static/sb-preview
4.0K    ./storybook-static/sb-addons/essentials-measure-6
 12K    ./storybook-static/sb-addons/ui-storybook-11
 12K    ./storybook-static/sb-addons/essentials-viewport-4
4.0K    ./storybook-static/sb-addons/links-0
184K    ./storybook-static/sb-addons/interactions-8
 84K    ./storybook-static/sb-addons/designs-10
4.0K    ./storybook-static/sb-addons/essentials-outline-7
8.0K    ./storybook-static/sb-addons/essentials-toolbars-5
 12K    ./storybook-static/sb-addons/essentials-backgrounds-3
 40K    ./storybook-static/sb-addons/essentials-actions-2
1.6M    ./storybook-static/sb-addons/storysource-9
380K    ./storybook-static/sb-addons/essentials-controls-1
2.4M    ./storybook-static/sb-addons
204K    ./storybook-static/sb-common-assets
3.4M    ./storybook-static/sb-manager
 19M    ./storybook-static/assets
 26M    ./storybook-static
```

A drop of 7M! (21%)

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests
- [x] benchmarking

#### Manual testing

I manually tested this on my local machine, by running `yarn storybook:ui:build` in the `code` directory.
Then I investigated the output of the `static-storybook` directory, and verified there are no `.map` files, except the ones that vite generated.

(We keep the sourcemaps generated by vite, because they are for the preview, and DO actually help the user debug their OWN code better.
The user can chose to disable sourcemaps themselves to get a faster build/deploy)

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [x] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [x] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/lib/cli/src/dev.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ function printError(error: any) {
   npmLog.heading = '';
 
   if (error instanceof Error) {
+    console.log('instanceof Error');
```
Comment (yannbf) [inline code/lib/cli/src/dev.ts:13]:
```suggestion
```

File: code/lib/cli/src/dev.ts [hunk 2]
```diff
@@ -18,6 +19,7 @@ function printError(error: any) {
       logger.error(error as any);
     }
   } else if (error.compilation?.errors) {
+    console.log('else');
```
Comment (yannbf) [inline code/lib/cli/src/dev.ts:22]:
```suggestion
```

File: code/lib/core-server/src/build-dev.ts [hunk 3]
```diff
@@ -157,7 +157,9 @@ export async function buildDevStandalone(
       );
 
     // eslint-disable-next-line no-console
+    console.log('VVVVVV problems VVVVVV');
```
Comment (yannbf) [inline code/lib/core-server/src/build-dev.ts:160]:
```suggestion
```

File: code/lib/core-server/src/build-dev.ts [hunk 4]
```diff
@@ -157,7 +157,9 @@ export async function buildDevStandalone(
...
     console.log(problems.map((p) => p.stack));
+    console.log('^^^^^^ problems ^^^^^^');
```
Comment (yannbf) [inline code/lib/core-server/src/build-dev.ts:162]:
```suggestion
```

**Review by yannbf — APPROVED** [review]
LGTM!

**Review by JReinhold — APPROVED** [review]
🗜️

**Comment by yannbf** [comment]
> 🗜️

more like
image

**Pull request merged**